src/formatDate.js

```diff
@@ -49,8 +49,43 @@
  * @returns {string}
  */
 
+// formatDate('97/02/18', ['YY', 'MM', 'DD', '/'], ['DD', 'MM', 'YYYY', '.'])
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prevDate = date.split(fromFormat[3]);
+  const prevDay = fromFormat.indexOf('DD');
+  const prevMonth = fromFormat.indexOf('MM');
+  let prevYear = fromFormat.indexOf('YY');
+  const newDay = toFormat.indexOf('DD');
+  const newMonth = toFormat.indexOf('MM');
+  let newYear = toFormat.indexOf('YY');
+  const newDate = [];
+
+  if (prevYear === -1) {
+    prevYear = fromFormat.indexOf('YYYY');
+  }
+
+  if (newYear === -1) {
+    newYear = toFormat.indexOf('YYYY');
+  }
+
+  newDate[newDay] = prevDate[prevDay];
+  newDate[newMonth] = prevDate[prevMonth];
+  newDate[newYear] = prevDate[prevYear];
```

Comment:
```suggestion
  [newDate[newDay], ...] = [prevDate[prevDay], ...];
```
you can use destructuring for this case